core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -99,7 +104,7 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
         }
       })
 
-    Await.result(result.future, Duration.Inf)
+    Await.result(result.future, timeout seconds)
```

Comment:
FYI, this is probably going to generate compiler warning about "postfix ops", so I'd add a `import scala.language.postfixOps` to fix them.